lighthouserc.js

```diff
@@ -1,33 +1,59 @@
+const path = require('path');
+const fs = require('fs');
+
+const getURLs = () =>
+    fs
+        .readdirSync('src/pages', { withFileTypes: true })
+        .filter(
+            dirent =>
+                dirent.isFile() &&
+                !(dirent.name.startsWith('_') || dirent.name.startsWith('[')) &&
+                !dirent.name.includes('index')
+        )
+        .map(
+            dirent =>
+                `http://localhost:3000/developer/${
+                    path.parse(dirent.name).name
+                }`
+        );
 module.exports = {
     ci: {
         collect: {
-            staticDistDir: './.next',
             numberOfRuns: 2,
-            url: ['http://localhost:3000/server/pages/index.html'],
+            startServerCommand: 'yarn start',
+            url: ['http://localhost:3000/developer', ...getURLs()],
+            settings: {
+                preset: 'desktop',
+            },
         },
         upload: {
             target: 'temporary-public-storage',
         },
         assert: {
             preset: 'lighthouse:no-pwa',
             assertions: {
-                'link-text': 'warn',
-                'link-name': 'warn',
+                'aria-allowed-attr': 'warn',
+                canonical: 'warn',
                 'csp-xss': 'warn',
-                'aria-roles': 'warn',
+                'color-contrast': 'warn',
                 'errors-in-console': 'warn',
-                'html-has-lang': 'warn',
-                'image-alt': 'warn',
-                'unsized-images': 'warn',
-                'crawlable-anchors': 'warn',
+                'external-anchors-use-rel-noopener': 'warn',
+                'heading-order': 'warn',
+                'inspector-issues': 'warn',
                 'offscreen-images': 'warn',
+                'non-composited-animations': 'warn',
+                'unsized-images': 'warn',
+                'unused-css-rules': 'warn',
                 'unused-javascript': 'warn',
-                label: 'warn',
-                'external-anchors-use-rel-noopener': 'warn', // Should reset this when we get a solution.
-                'heading-order': 'warn',
-                'categories:performance': ['error', { minScore: 0.8 }],
-                'categories:accessibility': ['error', { minScore: 0.7 }],
-                'categories:best-practices': ['error', { minScore: 0.8 }],
+                'uses-responsive-images': 'warn',
+                'valid-source-maps': 'warn',
+                'link-name': 'warn',
+                'link-text': 'warn',
+                'aria-roles': 'warn',
+                'uses-rel-preconnect': 'warn',
+                'categories:performance': ['error', { minScore: 0.55 }],
```

Comment:
I believe this is mainly because the tests are using `next start` now in order to dynamically generate the pages to test. Previously it was just opening the `index.html` file--which I think makes sense that loading a static HTML file has better Lighthouse performance than retrieving those assets from the next server. Not sure where or how Github actions actually is running the next server, but I'm assuming the performance isn't optimal like it would be in prod.

Also I think some of the pages have slightly worse performance than the main page, and this minScore has to reflect the worst performing page.

Reply:
Makes sense. I agree and do think other pages besides the main page are less performant-- at least we'd be to track it now.